### PR TITLE
leaderboard Include exclude and selection

### PR DESCRIPTION
**Pull request by bcolloran**
closes #978

Builds off of @AdityaHegde's #877 PR to update the styling and components to the latest design discussed by me and @hamilton  

Still WIP, though most of the proposed leaderboard design is in place save for the dots menu replacing the filter mode toggle icon (and there are a few polish items I'm looking at).

Have not touched the filter pill dropdown menus pending design sketches or more discussion about how we want search to fit in there.

### CODE REVIEW

**Comment by bcolloran** [comment]
@hamilton this is ready for an experience review of the core include/exclude design -- it's implemented based on the existing mocks since we haven't discussed it further yet. Having played with it live it does read largely as I'd imagined, so for me it feels pretty clear what things are selected and what things are not, and which items are included/excluded from the filter set. I won't argue it's perfect of course, so LMK if you have brainstorms for improvements, but I think we could go live with this design (modulo the dots menu we discussed)

**Comment by magorlick** [comment]
Thanks for your work here, @bcolloran! I am really enjoying this feature - it is nice to be able to exclude outlier segments without clicking around a lot.  

I think that many of the design mocks I have seen focus on the way the leaderboard looks and feels, but I felt confused when I was interacting with adjacent content in the filter pills and the detail board. A few notes on the experience below that might help.

The tool tip language is a bit confusing.  Can we update the copy?
image

Copy: 
"Output includes selected values" /  toggle to exclude values ... Click"
"Output excludes selected values" /  toggle to include values ... Click"


When we are excluding, can we make the checks "X" as we see in the leaderboard for consistency.  We should consider making the checkmarks here more similar to those in the leaderboard.

image
image


I lose my sense of what is toggled when I click into the detail table when I have an exclude filter on.
image

- Can we put check marks and x's next to the value names that are included or excluded, respectively, and grey them out inside the detail table?
- Can we put the toggle inside the detail table to the right of "<- All Dimensions"?

**Comment by bcolloran** [comment]
@AdityaHegde, this builds off of your PR #877 , so I think you will be the best person to review it. There are mostly changes in the  components, not the internal logic you built in your PR which I mostly left alone. Also, @djbarnwal reviewed the stuff having to do with the dimension table in #996, so you don't need to spend a ton of time on that section.

@hamilton and @magorlick -- imo, usabilty-wise I think this is pretty good and able to be merged. I don't really love how the dots menu toggle in the leaderboard feels, nor how the include/exclude button in the dimension detail table feels. I think that a [switch UI element like this](https://mui.com/material-ui/react-switch/) might be a nice way to indicate this toggle state (could have a switch as a line item in a menu and a switch in a button, which would clarify the classic UI ambiguity that this toggle falls into between whether an item is being used to indicate the current state or the state that you will enter upon interacting with the item. Very open to other brainstorms, let's discuss. 

Meantime, however, I think the current interaction is not ideal, but is good enough, and that we should merge this and follow up in a separate PR to get smaller details like that, and that we should lock this down so that @djbarnwal can integrate the handful of changes made here into his filter pill menu work for the search stuff.

File: web-local/src/lib/components/chip/core/Chip.svelte
Comment (AdityaHegde) [inline web-local/src/lib/components/chip/core/Chip.svelte:18]: We should use the default colours from chip-types here.

File: web-local/src/lib/components/leaderboard/LeaderboardHeader.svelte
Comment (AdityaHegde) [inline web-local/src/lib/components/leaderboard/LeaderboardHeader.svelte:17]: nit: filterKey and otherFilterKey are unused.

File: web-local/src/lib/components/leaderboard/LeaderboardHeader.svelte
Comment (AdityaHegde) [inline web-local/src/lib/components/leaderboard/LeaderboardHeader.svelte:14]: We should use dispatch instead of passing callbacks. Please change in other components as well.

**Comment by bcolloran** [comment]
thanks @AdityaHegde, all comments are addressed.

@magorlick code review for this is done, and it's ready for another experience pass. Can be merged pending sequencing with the dimension search work

**Comment by magorlick** [comment]
Thank you for these changes @bcolloran !  Together with dimension search, we have a really powerful way to quickly explore what is going on for dimensions with lots of values.  Four notes, mostly polish bits.  Would be nice to get in before release, but if necessary we could ship this as is.

P2. I am not sure why, but this branch has changed the way that the side bar collapses where a slim grey bar remains on the left hand side. That is not true on main today.

https://user-images.githubusercontent.com/5587788/195237154-4b66f88a-8f3d-4916-9b86-765dcce1ca10.mp4

P2. The leaderboard toggle for include/exclude is not present for very long column names.

https://user-images.githubusercontent.com/5587788/195237254-18ae8e22-f7fa-4141-8871-00c1820f83d0.mp4

P2. When the name is just-so-long that it almost hits the edge of the leaderboard and you can hover over the toggle, the name of the column name shifts to the left to accommodate the icon.

https://user-images.githubusercontent.com/5587788/195237365-d881095f-8b29-4b5a-9645-578f15012898.mp4

P2. the copy for the leaderboard toggle hover over still doesn't feel intuitive when no filters are selected.  We might change this to something more like "click to exclude selected values / click to include selected values"
image

**Comment by hamilton** [comment]
This looks great! one minor issue: in the dimension table, if nothing is selected, I can't toggle between include and exclude.

**Comment by bcolloran** [comment]
Thanks for catching all those items @magorlick and @hamilton, i think this is going to be a nice feature. All of those items (as well as the styling things we discussed on slack, Hamilton) have been fixed, except for this one Marissa, which I could not reproduce:

> P2. I am not sure why, but this branch has changed the way that the side bar collapses where a slim grey bar remains on the left hand side. That is not true on main today.

Here is what I am seeing on this branch:
[Screencast 2022-10-12 11:48:00.webm](https://user-images.githubusercontent.com/2380975/195423835-9fb75417-c163-46d1-a8b9-665706907e70.webm)

**Comment by bcolloran** [comment]
@djbarnwal and @AdityaHegde: per our discussion this morning, I believe this is now ready for you guys to integrate with Dhiraj's work and the outstanding API changes. Thanks!

**Comment by djbarnwal** [comment]
@bcolloran @magorlick This branch is now updated with main. We can merge this if everything else checks out.

**Comment by magorlick** [comment]
experience lgtm - @bcolloran is up for the final pass before merging.

**Review by magorlick — APPROVED** [review]

**Pull request merged**
